File: coindex/src/main/java/com/ironhack/coindex/model/Position.java
package com.ironhack.coindex.model;

import javax.persistence.*;
import java.math.BigDecimal;
import java.util.Collection;

@Entity
public class Position {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String coinId;
    private BigDecimal amount;
    @OneToMany(mappedBy = "position")
    private Collection<PositionUpdate> updates;
    @ManyToOne
    private Portfolio portfolio;

    public Position() {
    }

    public Position(String coinId, BigDecimal amount, Collection<PositionUpdate> updates) {
        this.coinId = coinId;
        this.amount = amount;
        this.updates = updates;
    }


    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getCoinId() {
        return coinId;
    }

    public void setCoinId(String coinId) {
        this.coinId = coinId;
    }

    public BigDecimal getAmount() {
        return amount;
    }

    public void setAmount(BigDecimal amount) {
        this.amount = amount;
    }

    public Collection<PositionUpdate> getUpdates() {
        return updates;
    }

    public void setUpdates(Collection<PositionUpdate> updates) {
        this.updates = updates;
    }
}

File: frontend/src/app/models/position.ts
import { PositionUpdate } from "./position-update";

export class Position {
    // private Long id;
    // private String coinId;
    // private BigDecimal amount;
    // private List<PositionUpdateDTO> updates;

    constructor(
        private _id: number,
        private _coinId: string,
        private _amount: number,
        private _updates: PositionUpdate[],
    ) {}

    public get id(): number {
        return this._id;
    }

    public get coinId(): string {
        return this._coinId;
    }

    public get amount(): number {
        return this._amount;
    }

    public get updates(): PositionUpdate[] {
        return this._updates;
    }

    public set id(id: number) {
        this._id = id;
    }

    public set coinId(coinId: string) {
        this._coinId = coinId;
    }

    public set amount(amount: number) {
        this._amount = amount;
    }

    public set updates(updates: PositionUpdate[]) {
        this._updates = updates;
    }
}

